networkx/algorithms/covering.py

```diff
@@ -0,0 +1,131 @@
+    are covered.
+
+    Parameters
+    ----------
+    G : An undirected, unweighted graph.
+
+    Returns
+    -------
+    min_cover : set
+
+        The covering is returned as a set, ``min_cover``.
+        It contains all the edges of minimum edge cover
+        in form of tuples.
+
+    matching_algorithm : function
+        It specifies the maximum weight matching algorithm to be used.
+
+    Notes
+    -----
+    An edge cover of a graph is a set of edges such that every vertex of
+    the graph is incident to at least one edge of the set.
+    A minimum edge cover is an edge covering of smallest possible size.
+
+    Due to its implementation, the worst-case running time of this algorithm
+    is bounded by the worst-case running time of the function
+    `matching_algorithm`.
+
+    Minimum edge cover for bipartite graph can also be found using the
+    function present in :mod:`networkx.algorithms.bipartite.covering`
+    """
+    if nx.number_of_isolates(G) > 0:
+        # ``min_cover`` does not exist as there is an isolated vertex
+        raise nx.NetworkXException(
+            "Graph has a vertex with no edge incident on it, "
+            "so no edge cover exists.")
+    if matching_algorithm is None:
+        matching_algorithm = nx.max_weight_matching
+    maximum_matching = matching_algorithm(G)
+    # ``min_cover`` is superset of ``maximum_matching``
+    min_cover = set(maximum_matching.items())
+    # iterate for uncovered nodes
+    uncovered_nodes = set(G) - set(v for u, v in min_cover)
+    for v in uncovered_nodes:
+        # Since `v` is uncovered, each edge incident to `v` will join it
+        # with a covered node (otherwise, if there were an edge joining
+        # uncovered nodes `u` and `v`, the maximum matching algorithm
+        # would have found it), so we can choose an arbitrary edge
+        # incident to `v`. (This applies only in a simple graph, not a
+        # multigraph.)
+        #
+        # Since the graph is guaranteed to be undirected, we need only
+        # add a single orientation of the edge.
+        u = arbitrary_element(G[v])
+        min_cover.add((u, v))
+        min_cover.add((v, u))
+    return min_cover
+
+
+def is_edge_cover(G, cover):
+    r"""Decides whether a set of edges is a valid
+    edge cover of the graph.
+
+    Given a set of edges, whether it is an edge covering can
+    be decided if we just check whether all nodes of the graph
+    has an edge from the set, incident on it.
+
+    Parameters
+    ----------
+    G : An undirected, unweighted graph.
+    cover : set
+        Set of edges to be checked.
+
+    Returns
+    -------
+    bool
+        Whether the set of edges is a valid edge cover of the graph.
+
+    Notes
+    -----
+    An edge cover of a graph is a set of edges such that every vertex of
+    the graph is incident to at least one edge of the set.
+    """
+    return len(set(G) - set(x[0] for x in cover)) == 0
+
+
+def is_min_edge_cover(G, cover):
```

Comment:
What is the purpose of this function? Since this function computes `min_edge_cover` anyway, users would be better off just calling `min_edge_cover` themselves.